Ijod

temas opened the pull request:
This implemens ijod for core synclet results.

A few of the highlights now are:
- JSON data is tracked with a murmurhash3 to ensure we don't reprocess as much
- Writes from the synclets are batch processed leading to a large performance gain.

There are currently two primary TODOs before looking to merge.
- Look into data access patterns such as /getCurrent
- More testing!

Please test this branch and see how it feels.

quartzjer committed:
first pass at ijod conversion

quartzjer committed:
ijod is working for synclets, and cleans up so much tech debt and old code/logic in the process

quartzjer committed:
cleaned up some tech debt of copy pasta, shared web service requests now between synclets and push

quartzjer committed:
switched push to using ijod natively too

quartzjer committed:
fixing up tests and other teseting related fixes

quartzjer committed:
draft main migration tool

quartzjer committed:
big merge, fix conflict

quartzjer committed:
crazy merge conflict

quartzjer committed:
bad args, remaining merge fixing

quartzjer committed:
cleaning up tests to work fully now

quartzjer committed:
basic bulk migration tool

Commit:
Merge branch 'master' into ijod

Commit:
Fixes for ijod to use zlib

Commit:
Make sure we're using gzip compression, fully working on v0.6 and mongo dump working

Commit:
Merge branch 'master' into ijod

Commit:
Merge branch 'master' into ijod

Commit:
Moving to a new compress-buffer based IJOD

Commit:
Movign to compress-buffers is complete as well as a new smartBatchAdd

Commit:
Bit of cleanup on the add data processing

Commit:
Merge branch 'master' into ijod

othiym23 committed:
Let's dependency!

Commit:
Add the lost getCurrent interface

Commit:
Merge branch 'ijod' of github.com:LockerProject/Locker into ijod

temas commented:
getCurrent is unlost and functioning.

Commit:
Updating for an ijod migration and some related fixes

Commit:
Updating with master

Commit:
It sure helps to call your step callbacks

Commit:
A simple tool to pull down getCurrent of a few synclets for verification

Commit:
Updating to master

temas commented:
This is now ready to merge and start testing on our sandbox.  I'd appreciate any final reviews before I pull the trigger.

There is a verification tool in the Ops directory called checkCurrent.js.  It is ran such as: `node checkCurrent.js localhost` and to test against ijod use:  `node checkCurrent.js localhost ijod`.  These are using the same calls, just different file naming.  The list of synclets can be adjusted at the top of the file for testing based on available data sets.  The results are put in a verification directory from where it's ran.  There seems to be a small issue with trailing \n missing sometimes, but all of the data is the same, verified with sha1 sums and diff.

Commit:
Merge branch 'master' into ijod

Commit:
Few more cleanups I'm finding for archiving

Commit:
Revert "Few more cleanups I'm finding for archiving"

This reverts commit 2df12ab456c3ccfce127f9666cdc29106ce0dd7d.

Commit:
Use the explicit close handler so we don't keep files around too long.

Commit:
Merge branch 'master' into ijod

Commit:
Use default mongo options when running the test suite

Commit:
Revert "Use default mongo options when running the test suite"

This reverts commit 7735358d36db4d04bf0e86e63d5ead4c69a46feb.

Commit:
Merge branch 'master' into ijod

Commit:
Better error handling and rollback

This changes transactions so it has actual transaction rollback.  The
batch add defaults to using this with 5 attempts to land a commit.  I'm
going fail hard if those 5 attempts fail until I understand why it's
failing.

Commit:
Merge branch 'ijod' of github.com:LockerProject/Locker into ijod

Commit:
Merge branch 'master' into ijod

Commit:
Add "jenkins" rule and move custom jenkins stuff here

This stuff is much more transparent if we push it into the source tree,
rather than putting it in the jenkins config.  To the greatest extent
possible, Jenkins should just run "make jenkins" and let us take care of
the rest here.

Commit:
Disable gracefulfs and check for double closes

Commit:
Merge branch 'ijod' of github.com:LockerProject/Locker into ijod

Commit:
Merge branch 'ijod' of github.com:LockerProject/Locker into ijod

Commit:
Merge branch 'master' into ijod

Commit:
Merge branch 'master' into ijod

Commit:
self is actually this, make it so

Commit:
Merge branch 'ijod' of github.com:LockerProject/Locker into ijod

mdz commented on Ops/webservice-push.js:
```diff
@@ -27,36 +26,24 @@ module.exports = function(locker) {
         });
     });
 
-    // copy pasta from the synclet code, these should be utilizing some generic stuff instead
```
Mmmm, pasta

mdz commented on Ops/webservice-synclets.js:
```diff
@@ -0,0 +1,111 @@
+                }
+            }
+            if (file) {
+                var stream = fs.createReadStream(path.join(lconfig.lockerDir, lconfig.me, req.params.syncletId, 'photos', file));
```
Isn't there a res.sendFile or something like that?

Commit:
Merge branch 'master' into ijod

Commit:
Merge branch 'master' into ijod

Commit:
Give up

Commit:
Merge branch 'master' into ijod

Commit:
Fix push for the IJOD changes

Commit:
Merge branch 'ijod' of github.com:LockerProject/Locker into ijod

Commit:
Refactoring service manager scheduling to get rid of logic bombs.

There were known and unknown logic bombs in the code that have been
purged by this pass.  All scheduling is centralized and synclets are
only executed from one location now.

Commit:
Specific debugging trying to find a weird error

Commit:
typo in paging

Commit:
More specific debug logging
